lib/src/strings.dart

```diff
@@ -105,12 +105,31 @@ String _tryYamlEncodeFolded(String string, int indentation, String lineEnding) {
     result = '>-\n' + ' ' * indentation;
   }
 
+  /// If the line starts with a empty character (excluding indentation), the
+  /// newline character(\n) before and after the line will be remained.
+  var emptyBegin = false;
+  final replacedString =
+      trimmedString.replaceAllMapped(RegExp(r'\n(.*)'), (match) {
```

Comment:
Would it be better to do `.split('\n')`, not sure. Just asking if we've considered it?